client/human/HumanClientFSM.cpp

```diff
@@ -114,9 +114,13 @@ boost::statechart::result WaitingForSPHostAck::react(const HostSPGame& msg) {
 
 boost::statechart::result WaitingForSPHostAck::react(const Disconnection& d) {
     if (TRACE_EXECUTION) DebugLogger() << "(HumanClientFSM) PlayingGame.Disconnection";
-    //Note: Any transit<> transition must occur before the MessageBox().
-    // MessageBox blocks and can allow other events to transit<> to a new state
-    // which makes this transit fatal.
+    // Note: All transitions, transit<> or disable_event(), must occur in the order
+    // that they cause changes to the state machine.
+    // - MessageBox blocks the UI, but continues handling events so other events
```

Comment:
MessageBox handles events? Or the FSM continues handling events while MessageBox has the UI blocked? Why does a UI block matter to the FSM?